src/dune/build_system.ml

```diff
@@ -1420,28 +1484,38 @@ end = struct
     in
     let* () =
       if rule_need_rerun then (
-        List.iter targets_as_list ~f:(fun target ->
-            Cached_digest.remove (Path.build target);
-            Path.unlink_no_err (Path.build target));
         let from_Cache =
           match (do_not_memoize, t.caching) with
           | true, _
           | _, None ->
             None
           | false, Some { cache = (module Caching) as cache; _ } -> (
             match Caching.Cache.search Caching.cache rule_digest with
-            | Ok (_, files) -> Some (files, cache)
+            | Ok (_, files) ->
```

Comment:
Please check if my understanding is correct:

At this point we hope that our earlier `hint` has resulted in the cache being populated with the right build artifacts, leading to the cache hit. 

If that's correct, I think adding the above sentence as a comment here would be useful for future readers.

Reply:
This is the "local" cache, which just pulls whatever is available from disk, it has no dependency on the distributed cache and does not concern itself with it. If no distributed cache is present, the local cache with happily promote things to disk and fetch them later.

The dependency is the other way around : some mechanisms may try to populate the local cache. Right now there's the hint message, and the per-commit indexing.

Reply:
Thanks! In the first round of review, I somehow missed that this change only adds logging.